src/controllers/activegate/reconciler/statefulset/statefulset_test.go

```diff
@@ -196,20 +199,43 @@ func TestStatefulSet_Volumes(t *testing.T) {
 func TestStatefulSet_Env(t *testing.T) {
 	instance := buildTestInstance()
 	capabilityProperties := &instance.Spec.ActiveGate.CapabilityProperties
-	deploymentMetadata := deploymentmetadata.NewDeploymentMetadata(string(testUID), deploymentmetadata.DeploymentTypeActiveGate)
+	deploymentMetadata := deploymentmetadata.NewDeploymentMetadata(testUID, deploymentmetadata.DeploymentTypeActiveGate)
 
 	t.Run(`without proxy`, func(t *testing.T) {
 		envVars := buildEnvs(NewStatefulSetProperties(instance, capabilityProperties,
 			testUID, "", testFeature, "MSGrouter", "",
 			nil, nil, nil,
 		))
-		assert.Equal(t, []corev1.EnvVar{
-			{Name: DTCapabilities, Value: "MSGrouter"},
-			{Name: DTIdSeedNamespace, Value: instance.Namespace},
-			{Name: DTIdSeedClusterId, Value: testUID},
-			{Name: DTDeploymentMetadata, Value: deploymentMetadata.AsString()},
+		assert.Equal(t, 7, len(envVars))
+		assert.Equal(t, envVars, []corev1.EnvVar{
```

Comment:
Please use `assert.Contains()` instead. Note that env variable sequence in this slice doesn't matter so you shouldn't assert that.

Reply:
Actually `assert.ElementsMatch` seems even better, switched to it.